src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts

```diff
@@ -99,7 +100,8 @@ class MLSConversationVerificationStateHandler {
       return this.logger.warn(`Epoch changed but conversation could not be found after waiting for 5 seconds`);
     }
 
-    if (!isMLSConversation(conversation)) {
+    const isSelfMLSConversation = isMLSConversation(conversation) && conversation.type() === CONVERSATION_TYPE.SELF;
```

Comment:
I don't even think you need to check for `isMLSConversation` at this point. checking for `self` is enough (proteus self conversation will already be filtered out on line 104)